packages/nextjs-demo/src/app/agent/api/route.tsx

```diff
@@ -59,15 +60,38 @@ CARAMEL MOCHA SPECIALTY LATTE $3.49
 MOCHA SPECIALTY LATTE $3.49
 `;
 
+/**
+ * The id of the Krispy Kreme corpus, from https://www.krispykreme.com
+ */
+const KK_CORPUS_ID = 'bd69dce6-7b56-4d0b-8b2f-226500780ebd';
+const MAX_CHUNKS = 4;
+
 class ClientMessage {
   constructor(public role: string, public content: string) {}
 }
 
-function ChatAgent({ conversation, model }: { conversation: ClientMessage[]; model: string }) {
+async function ChatAgent({
+  conversation,
+  model,
+  docs,
+}: {
+  conversation: ClientMessage[];
+  model: string;
+  docs?: number;
+}) {
+  let prompt = KK_PROMPT;
+  const query = conversation[conversation.length - 1].content;
+  if (docs && query) {
+    const corpus = new FixieCorpus(KK_CORPUS_ID);
+    const chunks = await corpus.search(query, { limit: MAX_CHUNKS });
+    prompt += '\nHere is some relevant information that you can use to compose your response:\n';
+    prompt += chunks.map((chunk) => chunk.chunk.content).join('\n');
+  }
+  console.log(`Prompt:\n${prompt}`);
```

Comment:
If you run with `AIJSX_LOG=debug`, you don't need to do custom logging like this, but I also recognize that that logging will be very noisy.